src/components/Coupon/index.jsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Button, Icon } from '@edx/paragon';
+
+import CouponDetails from './CouponDetails';
+
+import { isTriggerKey } from '../../utils';
+
+import './Coupon.scss';
+
+const triggerKeys = {
+  OPEN_DETAILS: [' ', 'Enter'],
+  CLOSE_DETAILS: [' ', 'Enter', 'Escape'],
+};
+
+class Coupon extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      detailsExpanded: false,
+      dimmed: false,
+    };
+
+    this.toggleCouponDetails = this.toggleCouponDetails.bind(this);
+    this.handleCouponKeyDown = this.handleCouponKeyDown.bind(this);
+  }
+
+  setCouponOpacity(dimmedStatus) {
+    this.setState({
+      dimmed: dimmedStatus,
+    });
+  }
+
+  closeCouponDetails() {
+    this.setState({
+      detailsExpanded: false,
+      dimmed: false,
+    });
+  }
+
+  toggleCouponDetails() {
+    const detailsExpanded = !this.state.detailsExpanded;
+
+    this.setState({
+      detailsExpanded,
+      dimmed: false,
+    });
+
+    if (detailsExpanded) {
+      this.props.onExpand();
+    } else {
+      this.props.onCollapse();
+    }
+  }
+
+  handleCouponKeyDown(event) {
+    const { detailsExpanded } = this.state;
+    if (!detailsExpanded && isTriggerKey({ triggerKeys, action: 'OPEN_DETAILS', key: event.key })) {
+      event.preventDefault();
+      this.toggleCouponDetails();
+    } else if (detailsExpanded && isTriggerKey({ triggerKeys, action: 'CLOSE_DETAILS', key: event.key })) {
+      event.preventDefault();
+      this.toggleCouponDetails();
+    }
+  }
+
+  renderExpandCollapseIcon() {
+    const { detailsExpanded } = this.state;
+    const iconClass = detailsExpanded ? 'fa-chevron-up' : 'fa-chevron-down';
+    const screenReaderText = detailsExpanded ? 'Close' : 'Open';
+    return (
+      <Button
```

Comment:
Why is this a button if it has no action and is not accessible (`tabIndex="-1"`)

Reply:
Yeah, I will be making this just an icon without the button. Same thing with the error icon which is still to come.